platforms/jvm/toolchains-jvm-shared/src/main/java/org/gradle/jvm/toolchain/internal/install/JavaToolchainProvisioningService.java

```diff
@@ -60,4 +61,10 @@ default String getFileName(URI uri, ExternalResource resource) {
         }
         return fileName;
     }
+
+    default String getUniqueFileName(URI uri, ExternalResource resource, JavaToolchainSpec spec) {
```

Comment:
```suggestion
    default String buildFileNameWithDetails(URI uri, ExternalResource resource, JavaToolchainSpec spec) {
```

I think this is doing too much work for a `get` prefix, and it's not really `unique`.